src/components/toast/components/toast-display/index.tsx

```diff
@@ -0,0 +1,56 @@
+import classNames from 'classnames'
+import { ToastColor, ToastDisplayProps } from './types'
+import CloseButton from '../close-button'
+import s from './toast-display.module.css'
+
+const IS_DEV = process.env.NODE_ENV !== 'production'
+
+/**
+ * Display component for use with a toast library,
+ * such as react-hot-toast.
+ */
+function ToastDisplay({
+  title,
+  description,
+  color = ToastColor.neutral,
+  icon,
+  children,
+  renderActions,
+  dismissSelf,
+}: ToastDisplayProps) {
+  /**
+   * In development, throw an error if a toast has
+   * no title and no description (except when using custom content).
+   */
+  if (IS_DEV && !title && !description && !children) {
+    throw new Error(
+      `Toast must be provided either a "title" or a "description", or you must use "children" to render custom content in the Toast. Please ensure toast has a non-empty string for its "title" or "description", or provide custom "children" content to render.`
+    )
+  }
+
+  const DescriptionElem =
+    typeof description === 'string' ? 'p' : description ? 'div' : null
+
+  return (
+    <div className={classNames(s.root, s[`color-${color}`])}>
+      {icon ? <div className={s.iconArea}>{icon}</div> : null}
+      <div className={s.contentArea}>
+        {title ? <p className={s.title}>{title}</p> : null}
+        {description ? (
+          <DescriptionElem className={s.description}>
+            {description}
+          </DescriptionElem>
+        ) : null}
+        {children ? children : null}
+        {typeof renderActions == 'function' ? (
+          <div className={s.actions}>{renderActions({ dismissSelf })}</div>
+        ) : null}
+      </div>
+      <div className={s.closeArea}>
+        <CloseButton onClick={dismissSelf} ariaLabel="Dismiss notification" />
+      </div>
+    </div>
+  )
```

Comment:
🚧 What are your thoughts on updating this to use variables instead of ternaries with `null` falsy branches? I think it'll make the markup easier to parse.

Here's what I'm thinking (includes `children` for now, but as noted in another comment I'm not sure we want to allow this yet):

```tsx
  let iconElement
  if (icon) {
    iconElement = <div className={s.iconArea}>{icon}</div>
  }

  let titleElement
  if (title) {
    titleElement = <p className={s.title}>{title}</p>
  }

  let descriptionElement
  if (description) {
    descriptionElement = <p className={s.description}>{description}</p>
  }

  let actionsElement
  if (typeof renderActions == 'function') {
    actionsElement = <div className={s.actions}>{renderActions({ dismissSelf })}</div>
  }

  return (
    <div className={classNames(s.root, s[`color-${color}`])}>
      {iconElement}
      <div className={s.contentArea}>
        {titleElement}
        {descriptionElement}
        {children}
        {actionsElement}
      </div>
      <div className={s.closeArea}>
        <CloseButton onClick={dismissSelf} ariaLabel="Dismiss notification" />
      </div>
    </div>
  )
```

Not sure I love the `Element` suffixes, maybe `Slot` would be a better one?

Reply:
This is a hard one for me! I definitely appreciate that ternaries don't feel great all the time... however, I think there's another side of the coin (very much an opinion), which is that I find the grouping of JSX elements into the return statement of a component makes it easier to see the elements _and_ structure of the component in one go:

<img width="1173" alt="CleanShot 2022-05-24 at 09 43 58@2x" src="https://user-images.githubusercontent.com/4624598/170050558-836ae630-ef4d-42ed-9c35-28d853d5c289.png">

For me, this is the "render the markup" area of the component, and I know I'll see all the `className` etc passed into this area.

We had [a Slack thread on this a while back](https://hashicorp.slack.com/archives/C02T80JTNS2/p1649872956699629), and I don't remember exactly where we landed (if we landed anywhere?), but I would really appreciate being able to use ternaries in JSX components in order to keep the JSX grouped. I think once I pare back the description functionality, they'll all be one-line ternaries as well. It seems hard to both keep the JSX grouped, and avoid ternaries completely... I suppose maybe the `RenderIf` helper component mentioned in that thread might be useful?